src/pyproject_hooks/_in_process/_in_process.py

```diff
@@ -112,7 +112,11 @@ def find_distributions(self, context=None):
             # Delayed import: Python 3.7 does not contain importlib.metadata
             from importlib.metadata import DistributionFinder, MetadataPathFinder
 
-            context = DistributionFinder.Context(path=self.backend_path)
+            if context is None:
+                context = DistributionFinder.Context(path=self.backend_path)
+            else:
+                args = dict(vars(context), path=self.backend_path)
+                context = DistributionFinder.Context(**args)
```

Comment:
I think this is good. Thank you.
I had a deeper look into the implementation of `MetadataPathFinder`, and it seems that it only ever looks into [`context.name` and `context.path`](https://github.com/python/cpython/blob/v3.9.21/Lib/importlib/metadata.py#L505), so, *if we want*, this could be further simplified to:

```suggestion
            name = context.name if context else None
            context = DistributionFinder.Context(name=name, path=self.backend_path)
```

Reply:
I wonder if there is any type of regression test that could be added.